src/udisks2monitor.cpp

```diff
@@ -424,7 +424,7 @@ void UDisks2::Monitor::updatePartitionStatus(const UDisks2::Job *job, bool succe
                 }
             } else {
                 partition->activeState = QStringLiteral("failed");
-                partition->status = operation == UDisks2::Job::Mount ? Partition::Mounted : Partition::Unmounted;
+                partition->status = operation == UDisks2::Job::Mount ? Partition::Unmounted : Partition::Mounted;
```

Comment:
it might be worth checking the status in the ui component or fixing the error signal instead of replacing the status, as there may be errors in the future.

Reply:
It's not a problem of the ui component. Really

Reply:
@pvuorela  What do you think about this?

Reply:
Yea I think the UI should be fine, using the partition status the model has, but the status getting wrong.

Think the change here makes sense and fixes the problem, though I'm also wondering about the whole setup. There's also startMountOperation() following the action success and emitting reverting signals, but those ones don't seem to affect this case. 

cc @chriadam in case you have some idea